src/app/task/AddVendorSelection.tsx

```diff
@@ -0,0 +1,106 @@
+import { useGetEventQuery, useUpdateEventMutation } from '@/redux/api/eventApi'
+import { useCreateTaskMutation } from '@/redux/api/taskApi'
+import { CreateTaskRequest } from '@/types/task'
+import { Box } from '@mui/material'
+import React from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+
+interface TaskCreateFormProps {
+  eventId: string
+  setIsVendorVisible: (isVisible: boolean) => void
+}
+
+export const AddVendorSelection = ({
+  eventId,
+  setIsVendorVisible,
+}: TaskCreateFormProps) => {
+  const { data } = useGetEventQuery(eventId)
+
+  const [createTask] = useCreateTaskMutation()
+
+  const [updateEvent] = useUpdateEventMutation()
+
+  const [isVisible, setIsVisible] = React.useState(true)
+
+  const defaultValuesVendor: CreateTaskRequest = {
```

Comment:
There seems to be some duplication between `AddVenueSelection` and `AddVendorSelection` components. These components have a similar structure and functionality. Consider refactoring them to reduce duplication and improve maintainability.

Reply:
I appreciate your suggestion about refactoring the `AddVenueSelection` and `AddVendorSelection` components. However, at this stage, with only two templates, creating a separate template component might not provide significant benefits. Refactoring would require modifying all the test cases, which is time-consuming. Moreover, it could result in a similar number of lines of code due to the unique aspects of each component. If we plan to add more templates in the future, creating a template component would indeed be beneficial for maintainability and reducing duplication. We can reconsider this approach when that need arises.

Reply:
I’d also like to add that our current **code duplication metric** on SonarCloud is at **0%**. Hence, refactoring at this stage may not significantly enhance our code quality or maintainability.

Reply:
Yeah, i think its fine "for now" to have such duplications if there's time constraint. But then again, reducing duplications is a good practice to keep. I guess you can try to refactor it when you have enough time for that.

Reply:
Will surely do!